### ISSUE TITLE
Empty/NULL value as a part of the key when comparing query and csv does not work

### ISSUE THREAD

**Issue by fefeslon** [Seddryck/NBi#464]
Hey Cédric,

I have some trouble when trying to compare query and csv file where:

-  I need to use more columns as a (combined) key 

- There can be NULL/empty values (in some of the columns).

Example given:

**csv:**
```
id1;id2
1;0089
2;0090
2;
4;1275
```
**test:**
```
  <system-under-test>
    <execution>
      <query connectionString="@xyz" timeout-milliSeconds="120000"><![CDATA[
        SELECT 1, '0089'
        UNION
        SELECT 2, '0090'
        UNION
        SELECT 2, NULL
        UNION
        SELECT 4, '1275'
      ]]></query>
    </execution>
  </system-under-test>
  <assert>
    <equalTo>
      <column index="0" role="key" type="text" />
      <column index="1" role="key" type="text" />
      <resultSet file="csv\test.csv" />
    </equalTo>
  </assert>
```
**csv profile in the nbiset file:**
```
<csv-profile first-row-header="true" empty-cell="NULL" missing-cell="NULL" />
```

I have tried some other combinations:
```
<csv-profile first-row-header="true"/> --and changed the NULL value in the SELECT to ''
<csv-profile first-row-header="true" empty-cell="(null)" missing-cell="(null)" />
```
The result I got (same for all the combinations mentioned above):
![image](https://user-images.githubusercontent.com/25638272/54931320-0c42ec80-4f19-11e9-9c31-487419d75452.png)

Is there any specific reason why this does not work for keys (the behaviour is ok for the value columns)?

NOTE: YES, I am able to make a workaround if I add any real value to the csv profile (e.g. empty-cell = 1) and rewrite the query to replace NULL values (e.g. with the value 1), but that is some more manual work to do when preparing the data (setting the profile is quite easy but the queries have to be rewritten with that in mind).

Could this be somehow easily fixed/included, please?

Cheers.

**Comment by Seddryck** [comment]
Sounds like a bug, probably more something wrong when comparing keys than reading and using the csv profile. Will take a look.

**Comment by fefeslon** [comment]
Ok, thanks for the answer.

**Comment by Seddryck** [comment]
This [alpha release](https://ci.appveyor.com/project/Seddryck/nbi/builds/23352182) should manage the case for ```null``` and ```empty``` when the key column is of type ```text```.  

I'll provide a proper fix in v1.21 (including also ```null``` handling for ```numeric```, ```dateTime``` and ```boolean```).

**Comment by fefeslon** [comment]
Hey, the fix works.

Thanks!

**Comment by Seddryck** [comment]
Beta available in 1.21.0-beta0326

**Comment by Seddryck** [comment]
The merge in beta is not working ... do not use it.

**Comment by Seddryck** [comment]
Fixed in ```1.21.0-beta0329``` available at https://ci.appveyor.com/project/Seddryck/nbi/builds/27251471/artifacts
